share_app/src/main/java/org/odk/share/tasks/UploadJob.java

```diff
@@ -310,11 +312,31 @@ private void sendInstances(List<String> instanceIds, int progress, int total) {
                 dos.writeInt(c.getCount());
                 c.moveToPosition(-1);
                 while (c.moveToNext()) {
+                    String displayName = c.getString(
+                            c.getColumnIndex(InstanceProviderAPI.InstanceColumns.DISPLAY_NAME));
+                    String submissionUri = c.getString(
+                            c.getColumnIndex(InstanceProviderAPI.InstanceColumns.SUBMISSION_URI));
+
+                    dos.writeUTF(displayName);
+
+                    if (submissionUri == null) {
+                        dos.writeUTF("-1");
+                    } else {
+                        dos.writeUTF(submissionUri);
+                    }
+
+                    rxEventBus.post(new UploadEvent(UploadEvent.Status.UPLOADING, ++progress, total));
```

Comment:
Progress should be updated after the uploading is done. So after `sendInstance`, right?